core/src/main/java/nu/yona/server/crypto/CryptoSession.java

```diff
@@ -136,36 +129,34 @@ public static CryptoSession getCurrent()
 		return cryptoSession;
 	}
 
+	/**
+	 * Encrypts the given plaintext bytes.
+	 * 
+	 * @param plaintext the bytes to be encrypted
+	 * @return the encrypted bytes, with the crypto variant number prepended to it.
+	 */
 	public byte[] encrypt(byte[] plaintext)
 	{
-		try
-		{
-			return getEncryptionCipher().doFinal(plaintext);
-		}
-		catch (IllegalBlockSizeException | BadPaddingException e)
-		{
-			throw CryptoException.encryptingData(e);
-		}
+		return CryptoUtil.encrypt(CURRENT_CRYPTO_VARIANT_NUMBER, getEncryptionCipher(), plaintext);
 	}
 
+	/**
+	 * Decrypts the given ciphertext bytes.
+	 * 
+	 * @param ciphertext the bytes to be decrypted, with a leading crypto variant number
+	 * @return the decrypted bytes
+	 */
 	public byte[] decrypt(byte[] ciphertext)
 	{
-		try
-		{
-			return getDecryptionCipher().doFinal(ciphertext);
-		}
-		catch (IllegalBlockSizeException | BadPaddingException e)
-		{
-			throw CryptoException.decryptingData(e);
-		}
+		return CryptoUtil.decrypt(CURRENT_CRYPTO_VARIANT_NUMBER, getDecryptionCipher(), ciphertext);
 	}
 
-	private SecretKey getSecretKey(char[] password)
+	private static SecretKey getSecretKey(String password)
 	{
 		try
 		{
-			SecretKeyFactory factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA256");
-			KeySpec spec = new PBEKeySpec(password, SALT, 65536, 128);
+			SecretKeyFactory factory = SecretKeyFactory.getInstance(SECRET_KEY_ALGORITHM);
+			KeySpec spec = new PBEKeySpec(password.toCharArray(), SALT, 65536, SECRET_KEY_LENGTH_BITS);
```

Comment:
Why do you use so many (65536) iterations? Some recommendations on the number are [here](http://security.stackexchange.com/questions/3959/recommended-of-iterations-when-using-pkbdf2-sha256), but I think we should make this less because it is used very often and the password has a high entropy. Would a low number of iterations (e.g. 100) not be enough?